src/tiered.js

```diff
@@ -43,18 +43,14 @@ class TieredDatastore {
     throw Errors.notFoundError()
   }
 
-  has (key) {
-    return new Promise(async (resolve) => {
-      await Promise.all(this.stores.map(async (store) => {
-        const has = await store.has(key)
-
-        if (has) {
-          resolve(true)
-        }
-      }))
+  async has (key) {
+    for (const s of this.stores) {
```

Comment:
Maybe `Promise.race` here?  The previous code looked through all nested stores in parallel, this makes it serial but it should never have waited for them all to complete before resolving to a value.

Reply:
race is weird (http://bluebirdjs.com/docs/api/promise.race.html), with https://github.com/sindresorhus/p-any we would need to try catch twice to return a boolean.
do you think its worth it ? this looks nice and simple

Reply:
The semantics are a little convoluted.  Let's merge this then we can look at refactoring it if it proves to be a bottleneck.